### ISSUE TITLE
[TRACKER] Main bugs to fix before first official 3.0-alpha build

### ISSUE THREAD

**Issue by akien-mga** [godotengine/godot#8805]
This tracker issue is meant to list the main bugs that should be handled in priority before we can release a public alpha build.

That alpha build is not meant to be bug free, but these objectives should be met:
- The renderer should *work* (at least run, not all features need to be perfect) on most hardware supporting OpenGL 3.3 (desktop).
- OSX, Windows and X11 platforms should compile and run. It's OK for the first alpha if the other platforms don't work yet.
- The editor should be usable, i.e. the basic features should work as expected. The new theme should be polished for a better usability, as it currently has many outstanding usability regressions.
- Rendering features advertised in the progress reports should work. They don't have to be perfect, but they should be usable so that testers can start seeing how the 3.0 will be like (and give feedback to improve it).
- The various outstanding bugs linked to compatibility breakage should be fixed or at least mitigated.
- C# integration is not a requirement for that build. If it's ready, that's great, but it's not a blocker.
- The rendering features planned for 3.0 and not implemented yet can be left for *after* that first build. There are enough outstanding issues to fix before adding new ones.

------

## High priority bugs

- [ ] Asset Library should be able to filter plugins by compatibility with engine version #7147
- [x] Removing a rigidbody and re-adding it to the scene results in errors and the rigidbody disappearing #10074
- [ ] Graphic artifacts in 3d scenes. #10149
- [x] Glow post effect is broken for softlight and screen blend modes (regression from 5567e89) #9106
- [x] Font hinting causing chunky fonts #9441
- [x] Godot freeze computer, when shadow contact set to 1 #9740
- [x] Glow seems to be broken (regression from 772485c) #9797 
- [x] Refraction is broken (regression from 772485c) #9803

------

*Edit:* Previous list, archived on 24 July.

**High priority bugs**

- [x] InputEvent's is_action{,_pressed,_released} don't work #8914 
- [x] Keyboard in editor unusable in latest code #8894
- [x] Image -> *.stex conversion is broken since it was made a Resource. #8801
- [x] Names of ClassDB-bound methods changed #8742
- [ ] OBJ files and DAE (better dae) are not imported correctly. Absent in .import folder. #8741
- [x] Broken lighting on Collada mesh with normal map [Intel HD 3000] #8711
- [ ] SSAO graphical issue with glitchy lines #8708
- [x] Using a texture in 3D always forces a re-import and makes it darker. #8684 
- [x] Various undefined shader problems #8681
- [x] Color degradation with directional shadow #8683
- [x] Normal maps in skinned/rigged meshes fail to work correctly #8673
- [x] Massive lag and OpenGL error spam on Windows with Intel graphics #8665 
- [x] Issues with the "View" menu in 3D mode #8489
- [x] Crash on AudioPlayer save scene #8596
- [x] color picker doesn't display colors properly #8372
- [x] The editor viewport with wrong size #8328
- [x] Crash making new shader in editor #8314
- [x] GI probes don't seem to have an effect #8194
- [x] Can apply CanvasItemMaterial to 3D Nodes #8121
- [x] Screen-space reflections issues #8119
- [x] Crash when opening default lighting config dialog #8082
- [x] TIME breaks canvas_item shaders #9123
- [x] Adding uniform sampler2D on shader_type canvas_item, brakes and forceclose godot #9143
- [x] Texture import freezes #9183

**Nice to fix** // can be for next alpha

- [x] Path editor plugin not working #8886
- [x] Strange effect with shadows enabled on DirectionalLight #8817
- [x] Particle shadows work only with directional light #8815
- [x] OmniLight can produce weird shadows when close to objects #8813
- [x] Accessing transform matrices data in shader breaks it. #8751
- [x] Varying in shader is not working #8749
- [x] Using INDEX on a particles shader, breaks the shader #8731
- [x] Viewport transparent background doesnt work #8703
- [x] Spotlights with shadows enabled cast light between mesh faces #8654
- [x] Collision shapes are drawn in wrong color and collision contact points are not drawn at all #8607
- [x] Light2D with shadows enabled near an occluder displays errors instead of shadows #8560
- [x] Glow affects gizmos #8485 
- [x] Object bounding boxes are always zero #8446
- [x] Thumbnails not visible in filesystem dock #8377
- [x] Only the first sampler showing up in ShaderMaterial Shader Param list #8353
- [x] SSAO doesn't play well with smooth shading #8237
- [x] Gizmos lose their colors #8118
- [x] Some 2D nodes aren't visible: canvas_item_add_triangle_array() doesn't work #7840 
- [x] FileSystem thumbnails display 2D scene (impressive bug!) #7812
- [x] Collision shapes and polygons not getting drawn on 2D #8033 (*bojidar-bg*)

**Comment by Corruptinator** [comment]
looking forward to see progress in 3.0!  :)

**Comment by tagcup** [comment]
#8886 and #8377 maybe? Also metallic workflow is not working properly, I'm not sure if there is an issue for that.

**Comment by n-pigeon** [comment]
Input class broken, probably after InputEvent rewrite #8914

**Comment by bojidar-bg** [comment]
I think #8931 is also good to have here.

**Comment by ficoos** [comment]
I think #8409 should also be here. Fixing this requires non trivial changes to `Variant` and also it changes the semantic of GDScript meaning that it's can't be fixed in a backward compatible way.

**Comment by bojidar-bg** [comment]
Added #8033 to the list, since it doesn't look good (and since we are getting lots of duplicates...)

**Comment by bojidar-bg** [comment]
Adding #9123, as it might crash the editor.

**Comment by DimitriyPS** [comment]
Why is this:
https://github.com/godotengine/godot/issues/8811
not included in the mandatory fixes to 3.0?

Godot 3.0 announced as a development engine for 3D. What if 3D basic physics does not work? You can forgive the outdated renderer, relatively poor performance... but the bug collisions kills any full project 3D games!

**Comment by akien-mga** [comment]
> Why is this:
#8811
not included in the mandatory fixes to 3.0?

It is, see its milestone. This tracker here is about the alpha, not about 3.0-stable.

**Comment by vnen** [comment]
Adding #9183, makes the editor quite unusable.

**Comment by rubenjavier** [comment]
@bojidar-bg https://github.com/godotengine/godot/issues/9143 also crashes the editor, and maybe its related to https://github.com/godotengine/godot/issues/9123 , since canvas_items shaders got broken

**Comment by vnen** [comment]
I think #9021 would be nice to fix.

**Comment by Zireael07** [comment]
#9254 is a physics regression, would be nice to see fixed before alpha so that I could actually test instead of wondering why do all cars end up belly up

**Comment by Corruptinator** [comment]
Would #9311 count as one of the bugs to fix? s3tc.stex could use a bit of a checkup because I found out that one of the slightly old 3.0 build where the Intel graphic lag and debug spam is fixed (https://github.com/godotengine/godot/tree/47b34bf79be1ba2fe4a99c8931732efac68a0b87)
actually has a fully functional, unbugged s3tc.stex file import process that does not leave the normalmap completely glitched.

Just pointing it out in case before the Alpha is uploaded.

**Comment by Grosskopf** [comment]
8711 seems fixed from my point of  view :)

**Comment by kubecz3k** [comment]
#9861 would be nice to have be fixed

**Comment by akien-mga** [comment]
> #9861 would be nice to have fixed

Not a bug :)

**Comment by Zireael07** [comment]
The list looks like it hasn't been updated in a month, and the vehicle physics thing still looks like a blocker...

**Comment by akien-mga** [comment]
> The list looks like it hasn't been updated in a month, and the vehicle physics thing still looks like a blocker...

Look better, it was updated yesterday. And that list is still only about the *alpha* stage, not all bugs are meant to be in it. All bugs with the 3.0 milestone will be reviewed before 3.0 is released.

**Comment by djrm** [comment]
@akien-mga i closed #9803

**Comment by tagcup** [comment]
How about #10074? Something is making angular velocity NaN in physics.

**Comment by HummusSamurai** [comment]
This should probably be here.
https://github.com/godotengine/godot/issues/10149

**Comment by HummusSamurai** [comment]
This was closed https://github.com/godotengine/godot/issues/9143 but it references https://github.com/godotengine/godot/issues/9263 as being the follow-up, not sure if you want to substitute it.

**Comment by akien-mga** [comment]
Closing as this tracker outlived its purpose (3.0-alpha1 is published already).

Let's see if we need another tracker for high priority issues in the master branch or if the 3.0 milestone is enough.
